### PR TITLE
MAINT: Drop python 3.7

### LINKED ISSUE #3057: MAINT: Drop support for python 3.7, following NEP 29
Port of https://github.com/dsgibbons/shap/issues/73

We should move towards [NEP 29 — Recommend Python and NumPy version support as a community policy standard](https://numpy.org/neps/nep-0029-deprecation_policy.html)

As explained in the NEP, there are several benefits of deprecating according to a predictable schedule, both for downstream users of the package and also for maintainers. The recommended drop schedule for python is:

```
On Dec 26, 2021 drop support for Python 3.7 (initially released on Jun 27, 2018)
On Apr 14, 2023 drop support for Python 3.8 (initially released on Oct 14, 2019)
On Apr 05, 2024 drop support for Python 3.9 (initially released on Oct 05, 2020)
On Apr 04, 2025 drop support for Python 3.10 (initially released on Oct 04, 2021)
On Apr 24, 2026 drop support for Python 3.11 (initially released on Oct 24, 2022)
```

It would be preferable to only drop one version per release. Suggested timeline for SHAP:
- 0.42.0 : last bugfix release for 3.7
- 0.43.0: drop support for 3.7
- 0.44.0: drop support for 3.8 (pending discussion)

Comparison with other packages (at time of writing):
- Numpy: 3.9-3.11
- Scipy: 3.9-3.11
- scikit-learn: `3.8-3.11`  (3.8 [will be dropped](https://github.com/scikit-learn/scikit-learn/issues/26438) on next release)

Checklist:

- [x] Make a prior release with a clear deprecation message
- [x] Update `requires-python` metadata
- [x] Update PyPI classifiers
- [x] Update CI test matrix

### PR DESCRIPTION
**Pull request by connortann**
Closes #3057

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Removed
 
+- Dropped support for python 3.7
```
Comment (thatlittleboy) [inline CHANGELOG.md]: Include the PR link + your GH username?

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,10 +19,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Add support for LightGBM ensembles containing single leaf trees in `TreeExplainer`
   ([#3094](https://github.com/slundberg/shap/pull/3094) by @thatlittleboy).
 
+### Fixed
+
```
Comment (thatlittleboy) [inline CHANGELOG.md]: I think this is a remnant from a merge conflict resolution?

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +36,7 @@ Patch release to provide wheels for a broader range of architectures.
   [#3083](https://github.com/slundberg/shap/pull/3083) by @connortann).
 - Fixed circular import issues with `shap.benchmark`
   ([#3076](https://github.com/slundberg/shap/pull/3076) by @thatlittleboy).
+- Dropped support for python 3.7
```
Comment (thatlittleboy) [inline CHANGELOG.md]: ditto here

**Review by thatlittleboy — APPROVED** [review]
Just a Couple of typos left over, please fix them, but I'll pre-approve!

File: CHANGELOG.md (hunk 2 above)
Comment (connortann) [reply CHANGELOG.md]: Ah good spot.

**Pull request merged**